File: Server/schemas/convalidation/convalidation_in.py
from pydantic import BaseModel, Field
from typing import Optional
from utils.constants import MAX_LENGTH_NAME, MAX_LENGTH_DESCRIPTION

class ConvalidationIn(BaseModel):
    id_student: int
    id_convalidation_type: int
    id_curriculum_course: int
    id_workshop: Optional[int] = None
    activity_name: Optional[str] = Field(None, max_length=MAX_LENGTH_NAME)
    description: Optional[str] = Field(None, max_length=MAX_LENGTH_DESCRIPTION)
    file_name: Optional[str] = Field(None, max_length=MAX_LENGTH_NAME)
    file_data: Optional[bytes] = None
    id_subject: Optional[int] = None
    id_department: Optional[int] = None 

File: Client/src/features/convalidations/types/convalidation_in.ts
export default interface ConvalidationIn {
  id_student: number
  id_convalidation_type: number
  id_curriculum_course: number
  id_workshop?: number
  id_activity_name?: number
  id_subject?: number
  description: string
  file_name?: string
  file_data?: string
  id_department?: number
}

